docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -25,184 +24,153 @@ participant "TLCC_Enclave Peer1_Org1" as TLCC_Enclave1 #99FF99
-    end note
-
-  Enclave1      -> Enclave1     :   SEK, CC_E_{PK,SK} <- keyGen
-    note left
-        - SEK denotes the state encryption key
-        - CC_E_{PK,SK} denotes chain code pub/priv key pair
-    end note
-  Enclave1      -> Enclave1     :   add SEK, CC_E_{PK,SK} to ecc_config
-  Enclave1      -> Enclave1     : SIG_ECC1 <- sign <CC_E_PK> with ECC1_S_SK
-    note left
-        Confirmation statement that the enclave has CC_E_SK.
-    end note
-
-  Enclave1      -> Enclave1     :   sealed_state <- Seal(ecc_config)
-  Enclave1      -> Enclave1     :   key_confirmation_message <- <ECC1_S_PK, SIG_ECC1, CC_E_PK>
+  Enclave1      -> Enclave1     :   SEK, Chaincode_DK <- keyGen()
+  Enclave1      -> Enclave1     :   extract Chaincode_EK from Chaincode_DK
+  note over Enclave1
+    - SEK denotes the state encryption key
+    - Chaincode_EK and Chaincode_DK denote the public and private encryption key pair
+  end note
+  Enclave1      -> Enclave1     :   Sig_Enclave1 <- sign Chaincode_EK with Enclave1_SK
 
-  return key_confirmation_message, sealed_state /'ECC1 -> Enclave1'/
-  return key_confirmation_message, sealed_state /'ECC_Manager1 -> ECC1'/
+  Enclave1      -> Enclave1     :   cckey_registration_message <- <Enclave1_VK, Sig_Enclave1, Chaincode_EK>
 
-  ECC_Manager1  -> Storage1     :   store sealed_state
+  return cckey_registration_message /'ECC1 -> Enclave1    '/
+  return cckey_registration_message /'E_Manager1 -> ECC1  '/
 
-  return key_confirmation_message /'Admin1 -> ECC_Manager1'/
+  return cckey_registration_message /'Admin1 -> E_Manager1'/
 
 end group
 
-'note right Admin1 #ff0000
-'    Discussion needed ... see below
-'end note
-
-'note right Admin1
-'    - Where do we store ECC1_S_PK, SIG_ECC, m, CC_E_PK??? At ECC or ERCC?
-'    - Do we treat key-mgnt as FPC lifcycle or is this application specific?
-'    - If FPC lifecycle => ERCC ; otherwise ECC
-'end note
-
-'note right Admin1
-'    **ERCC**
-'    + aligned with FPC lifcycle policy
-'    + central place where also FPC chaincodes store keys
-'    - how can ECC_Enclave access? m is stored in ERCC namespace
-'    - ERCC namspace can be accessed through cc2cc call
-'        but not protected through TLCC
-'    TBD ...
-'end note
-
-'note right Admin1
-'    **ECC**
-'    + easy to securely access m protected through TLCC
-'    - difficult with application policy
-'    TBD ...
-'end note
-
-group key confirmation
+group chaincode key registration
 
   note right of Admin1
-    Key confirmation is desirable in multi-org settings.
+    Key registration is desirable in multi-org settings.
     Assuming an org is running an enclave, other orgs may
     want to ensure that the private chaincode transaction
     can proceed in the case the running enclave is
     unavailable - for any possible or malicious cause.
-    Key confirmation can ensure that backup enclave are
+    Key registration can ensure that backup enclave are
     ready to take over and/or contribute.
   end note
```

Comment:
I am fine removing it.

Few things:
* single-org does not make sense for availability -- for obvious reasons. Why wouldn't it make sense for confidentiality/integrity in FPC?
* the comment aimed at clarifying that if you have multiple enclaves but one endorses, you don't know whether others can endorse until they either confirm the key or... endorse something.
* in the backup bullet, if a backup enclave confirms the key, then it received the private key and correctly extracted the public one.

Reply:
> * single-org does not make sense for availability -- for obvious reasons. 

actually, why? We talk about single-org, not single peer?  So you can still have as much availability as you want?

> Why wouldn't it make sense for confidentiality/integrity in FPC?

Why would you need enclaves (or a block-chain for that matter) for a permissioned single org config? In that case you anyway put all that trust into that single org so they could just make sure their server is secure and wouldn't really need sgx?  even, more easily probably implement everything based on a normal database and maybe stored procedures for chaincodes and could forgo fabric?

> * in the backup bullet, if a backup enclave confirms the key, then it received the private key and correctly extracted the public one.

Remember the function is called key registration. so nobody would think about apriori about key confirmation. And your scenario would only work after a key distribution also has happened.  In principle, the admin of the backup enclave should know it has the right (encrypted) key based on (necessary) trust in ERCC.  I guess a confirmation cannot hurt and can be done using the function as currently specified (but could also be done by necessary query function for the public, see other comment). But more pertinent for here, with the original text on backup enclave, it would have been very hard for a reader to connect the dots and understand what is meant without the explanation from this discussion?